### ISSUE TITLE
AcquireTokenAsync for UWP crashes on null redirectUri parameter

### ISSUE THREAD

**Issue by jevansaks** [AzureAD/azure-activedirectory-library-for-dotnet#1400]
**Which Version of ADAL are you using ?**
ADAL 4.3.0

**Which platform has the issue?**
UWP

**What authentication flow has the issue?**
* Desktop / Mobile
    * [x] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

Other? - please describe;

**Repro**

```csharp
  var authResult = await authCtx.AcquireTokenAsync(_resourceId, _clientId, null, platformParameters);
```

**Expected vs actual behavior**
Expected: it works.
Actual: it crashes.

The comments and code make it seem like I'm supposed to be able to pass in null for redirectUri and then have the code eventually call WebAuthenticationBroker.AuthenticateAsync without a redirectUri (implying that my app's ms-app:// URL will be used). 

However, what happens is that the code crashes immediately in AcquireTokenAsync after calling ValidateRedirectUri because it tries to dereference redirectUri.

https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/issues/96 (bullet 3 in the list) says that null is accepted for callbackUri, so it seems like it's expected to work.

**Possible Solution**
What I *think* is wrong is that ValidateRedirectUri is written as if the assignment to redirectUri will affect the caller, but it will not. This line: https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/e47ba877f3da1fbd7e3821e456cf33dcf26906c5/core/src/Platforms/uap/UapPlatformProxy.cs#L169

If I'm right, ValidateRedirectUri should be taking redirectUri by ref instead of by value and that would make the code work as written.

**Additional context/ Logs / Screenshots**
N/A

**Comment by bgavrilMS** [comment]
This should have been fixed and you can try it out with the nightly build https://www.myget.org/feed/aad-clients-nightly/package/nuget/Microsoft.IdentityModel.Clients.ActiveDirectory/4.3.1-internal32001dev

Passing NULL for redirect URI causes the redirectUri to be "http://sso" on UWP, which means that ADAL will try to use Web Broker. 

Closing this, fix will be available in next release - likely a minor release 4.3.1

**Comment by jevansaks** [comment]
Thanks!

**Comment by jennyf19** [comment]
Fixed in [ADAL v4.4.1](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/releases/tag/ADAL-v4.4.1)
